File: leetcode/2020/[Easy] best-time-to-buy-and-sell-stock-ii/Solution.scala
object Solution {
  def maxProfit(prices: Array[Int]): Int = {
    if (prices.length == 0) return 0
    var min = prices(0)
    var total = 0
    (1 until prices.length) foreach(i => {
      if (min < prices(i)) total += prices(i) - min;
      min = prices(i)
    })
    total
  }
}

File: leetcode/2020/[Easy] best-time-to-buy-and-sell-stock-ii/Solution.java
class Solution {
	public int maxProfit(int[] prices) {
		if (prices.length == 0) return 0;

		int profit = 0;

		int max = prices[prices.length - 1];
		for (int i = prices.length - 1; i >= 1; --i){
			if (prices[i] < prices[i - 1]) {
				profit += max - prices[i];
				max = prices[i - 1];
			}
		}
		profit += max - prices[0];
		return profit;
	}
}
